dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/page/HTMLPageAssetRenderedBuilder.java

```diff
@@ -138,22 +138,48 @@ public PageView build(final boolean rendered, final PageMode mode) throws DotDat
         final PageRenderUtil pageRenderUtil = new PageRenderUtil(
                 htmlPageAssetInfo.getPage(), systemUser, mode, language.getId(), this.site);
 
+        final Optional<Contentlet> urlContentletOpt = this.findUrlContentlet (request);
+
         if (!rendered) {
+
             final Collection<? extends ContainerRaw> containers =  pageRenderUtil.getContainersRaw();
-            return new PageView(site, template, containers, htmlPageAssetInfo, layout, canCreateTemplates,
-                    canEditTemplate, this.getViewAsStatus(mode, pagePersonalizationSet), pageUrlMapper, live);
+
+            return urlContentletOpt.isPresent()?
+                    new PageView(site, template, containers, htmlPageAssetInfo, layout, canCreateTemplates,
```

Comment:
PageView constructor has too many parameters. Could be refactored with a Builder